components/dialog/dialog-styles.js

```diff
@@ -0,0 +1,74 @@
+import { css } from 'lit-element/lit-element.js';
+
+export const dialogStyles = css`
+
+	:host {
+		display: none;
+	}
+
+	:host([opened]), :host([_state="showing"]), :host([_state="hiding"]) {
```

Comment:
These internal `_state` selectors ensure that `d2l-dialog` is not hidden in the DOM until after animation completed if the consumer has removed the `opened` attribute.